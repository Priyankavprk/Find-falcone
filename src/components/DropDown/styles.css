.menu {
  display: flex;
  flex-direction: column;
  height: 10vmin;
  padding: 5vmin;
  min-width: 5vmin;
}

.vehicleOptionsDisabled {
  color: #C7C8CB;
}

.container {
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
}

.value {
  display: flex;
  flex-direction: row;
}

.heading {
  padding: 2vmin
}
