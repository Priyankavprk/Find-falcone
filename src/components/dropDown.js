import React, { Component } from 'react';

import '../../node_modules/font-awesome/css/font-awesome.min.css';
import './style.css';

class Dropdownmenu extends Component {
  constructor() {
    super();
    this.state = {
      valueSelected: "Select",
      vehicles: [],
      selectedVehicle: null,
      availableVehicles: [],
      planetSelected: null
    }
    this.handlePlanetSelected = this.handlePlanetSelected.bind(this)
    this.setVehicle = this.setVehicle.bind(this)
  }

  handlePlanetSelected(event) {
    let planet = this.props.options.planets.filter((planet) => {
      if(planet.name === event.target.value) {
        return planet
      }
    })
    if(typeof this.props.handlePlanetSelected === 'function') {
      this.props.handlePlanetSelected(event.target.value, this.state.valueSelected)
      this.setState({
        valueSelected: event.target.value,
        planetSelected: planet
      })
    }
    if(this.state.selectedVehicle) {
      this.findTimeTaken()
    }
  }

  showPlanets() {
    if (this.props.options && this.props.options.planets) {
    let listItems = this.props.options.planets.map((planet, j) => {
      if(planet.name !== this.state.valueSelected && !this.props.destinationsSelected.includes(planet.name)) {
        if(j === 0) {
          return <option key={j} defaultValue="Select" disabled hidden value={planet.name} className="planetList">{planet.name}</option>
        }
        return <option key={j} value={planet.name} className="planetList">{planet.name}</option>
      } else if(planet.name === this.state.valueSelected) {
        return <option key={j} selected value={planet.name} className="planetList">{planet.name}</option>
      }
    })
    return listItems
  }
  }

  showVehicles() {
    if(this.props.options && this.props.options.vehicles) {
      let vehicleList = this.props.options.vehicles.map((vehicle, j) => {
        let count = []
        count = Object.values(this.props.usedVehicles).filter((v) => {
          if(vehicle.name === v) {
            return v
          }
        })
        if(vehicle.max_distance >= this.state.planetSelected[0].distance) {
          if(vehicle.total_no - count.length === 0 && vehicle.name !== this.state.selectedVehicle) {
            return (
              <label className= "vehicleOptionsDisabled" key={j}>
               <input type="radio" value={vehicle.name} name={`vehicle+${this.props.value}`} disabled/>
               {vehicle.name} ({vehicle.total_no - count.length})<br/>
              </label>
            )
          } else {
            return (
              <label key={j}>
               <input className= "vehicleOptions" type="radio" value={vehicle.name} name={`vehicle+${this.props.value}`}/>
               {vehicle.name} ({vehicle.total_no - count.length})<br/>
              </label>
            )
          }
        }
      })
      return vehicleList
    }
  }

  findTimeTaken() {
    let planet = this.props.options.planets.filter((data) => {
      if(data.name === this.state.valueSelected) {
        return data.distance
      }
    })
    let vehicle = this.props.options.vehicles.filter((data) => {
      if(data.name === this.state.selectedVehicle) {
        return data.speed
      }
    })
    this.props.handlePlanetSelected(null, null, planet[0].distance/vehicle[0].speed, this.props.value, vehicle)
  }

  setVehicle(event) {
    this.setState({
      selectedVehicle: event.target.value
    }, () => {
        this.findTimeTaken()
    })
  }

  render() {
    return (
        <div className="container">
        <div className="menu">
        <div className="heading">Destination {this.props.value}</div>
        <select id="planet" className="planetSelect" value={this.state.valueSelected} onChange={this.handlePlanetSelected}>
        {
         this.showPlanets()
        }
        </select>
        </div>
        {
          this.state.valueSelected && this.state.planetSelected &&
          <div onChange={this.setVehicle.bind(this)} className="values">
            {
              this.showVehicles()
            }
          </div>
        }
        </div>
    );
  }
}

export default Dropdownmenu;
