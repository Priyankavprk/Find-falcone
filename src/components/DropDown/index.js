import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { handlePlanetSelected, handleVehicleSelected, setTime } from '../../actions';
import '../../../node_modules/font-awesome/css/font-awesome.min.css';
import './styles.css';

export class Dropdownmenu extends Component {
  constructor() {
    super();
    this.state = {
      valueSelected: "Select",
      selectedVehicle: null,
      planetSelected: null
    }
    this.handlePlanetSelected = this.handlePlanetSelected.bind(this)
    this.setVehicle = this.setVehicle.bind(this)
  }

  handlePlanetSelected(event) {
    this.props.handlePlanetSelected(this.props.value, event.target.value)
    let planet = this.props.options.planets.filter((planet) => planet.name === event.target.value)
    this.setState({
      valueSelected: event.target.value,
      planetSelected: planet
    })
    if(this.state.selectedVehicle) {
      this.findTimeTaken()
    }
  }

  showPlanets() {
    if (this.props.options && this.props.options.planets) {
    let listItems = this.props.options.planets.map((planet, j) => {
      if(planet.name !== this.state.valueSelected && !(Object.values(this.props.destinationsSelected).includes(planet.name))) {
        if (j === 0) {
          return <option key={j} defaultValue="Select" disabled hidden value={planet.name} className="planetList">{planet.name}</option>
        }
        return <option key={j} value={planet.name} className="planetList">{planet.name}</option>
      } else if(planet.name === this.state.valueSelected) {
        return <option key={j} defaultValue value={planet.name} className="planetList">{planet.name}</option>
      }
      return null
    })
    return listItems
   }
  }

  showVehicles() {
    if(this.props.options && this.props.options.vehicles) {
      let vehicleList = this.props.options.vehicles.map((vehicle, j) => {
        let count = []
        count = Object.values(this.props.usedVehicles).filter((v) => vehicle.name === v)
        //Show vehicles that can cover the distance of planet selected
        if(vehicle.max_distance >= this.state.planetSelected[0].distance) {
          if(vehicle.total_no - count.length === 0 && vehicle.name !== this.state.selectedVehicle) {
            return (
              <label className= "vehicleOptionsDisabled" key={j}>
               <input type="radio" value={vehicle.name} name={`vehicle+${this.props.value}`} disabled/>
               {vehicle.name} ({vehicle.total_no - count.length})<br/>
              </label>
            )
          } else {
            return (
              <label key={j}>
               <input type="radio" value={vehicle.name} name={`vehicle+${this.props.value}`}/>
               {vehicle.name} ({vehicle.total_no - count.length})<br/>
              </label>
            )
          }
        }
        return null
      })
      return vehicleList
    }
  }

  findTimeTaken() {
    let planet = this.props.options.planets.filter((data) => data.name === this.state.valueSelected)
    let vehicle = this.props.options.vehicles.filter((data) => data.name === this.state.selectedVehicle)
    this.props.setTime(this.props.value, planet[0].distance/vehicle[0].speed)
  }

  setVehicle(event) {
    this.props.handleVehicleSelected(this.props.value, event.target.value)
    this.setState({
      selectedVehicle: event.target.value
    }, () => {
        this.findTimeTaken()
    })
  }

  render() {
    return (
        <div className="container">
        <div className="menu">
        <div className="heading">Destination {this.props.value}</div>
        <select id="planet" className="planetSelect" value={this.state.valueSelected} onChange={this.handlePlanetSelected}>
        {
         this.showPlanets()
        }
        </select>
        </div>
        {
          this.state.valueSelected && this.state.planetSelected &&
          <div onChange={this.setVehicle.bind(this)} className="values">
            {
              this.showVehicles()
            }
          </div>
        }
        </div>
    );
  }
}

Dropdownmenu.propTypes = {
  destinationsSelected: PropTypes.object,
  options: PropTypes.object,
  usedVehicles: PropTypes.object
}

const mapStateToProps = state => ({
  destinationsSelected: state.list.destinationsSelected,
  options: state.list.options,
  usedVehicles: state.list.usedVehicles,
})

const mapDispatchToProps = dispatch => ({
  handlePlanetSelected: (key, data) => dispatch(handlePlanetSelected(key, data)),
  handleVehicleSelected: (key, data) => dispatch(handleVehicleSelected(key, data)),
  setTime: (key, data) => dispatch(setTime(key, data)),
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Dropdownmenu)
