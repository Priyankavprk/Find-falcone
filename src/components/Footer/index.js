import React, { Component } from 'react';
import { Navbar } from 'react-bootstrap';

import './styles.css';

export class Footer extends Component {
  render() {
    return (
      <Navbar className="header">
        <Navbar.Brand className="title">Mission Find Falcone</Navbar.Brand>
      </Navbar>
    );
  }
}

export default Footer;
