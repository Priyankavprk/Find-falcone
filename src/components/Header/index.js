import React, { Component } from 'react';
import { Navbar } from 'react-bootstrap';

import './styles.css';

export class Header extends Component {
  render() {
    return (
      <Navbar className="header">
        <Navbar.Brand className="title">Finding Falcone!</Navbar.Brand>
      </Navbar>
    );
  }
}

export default Header;
