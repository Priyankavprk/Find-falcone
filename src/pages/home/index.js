import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { getData, getToken } from '../../actions';
import Footer from '../../components/Footer';
import Header from '../../components/Header';
import Dropdown from '../../components/DropDown';
import logo from '../../king.png';
import '../App.css';

export class Home extends Component {
  componentDidMount() {
    this.props.getData()
  }

  render() {
    if (this.props.options && this.props.options.planets) {
      return (
        <div className="App">
          <Header />
            <header className="App-header">
              <div className="image">
                <img className="icon" src={logo} alt="king"/>
              </div>
              <div className="titlemain">Select planet you want to search in </div>
              <div className="body">
                <Dropdown value={1}/>
                <Dropdown value={2}/>
                <Dropdown value={3}/>
                <Dropdown value={4}/>
                <div className="timeText">Time taken : {Object.values(this.props.data).reduce(function(a, b) { return a + b; }, 0)}</div>
              </div>
             <div>
                <input type="submit" onClick={() => this.props.getToken(this.props.history)} value="Find Falcone !" disabled={(Object.values(this.props.destinationsSelected).length === 4 && Object.values(this.props.usedVehicles).length === 4) ? false : true}/>
             </div>
             </header>
             <Footer />
          </div>
      );
    } else {
      return (
        <div className="loading"> Loading ... </div>
      )
    }
  }
}

Home.propTypes = {
  options: PropTypes.object,
  destinationsSelected: PropTypes.object,
  usedVehicles: PropTypes.object,
  data: PropTypes.object
}

const mapStateToProps = state => ({
  options: state.list.options,
  destinationsSelected: state.list.destinationsSelected,
  usedVehicles: state.list.usedVehicles,
  data: state.list.data,
})

const mapDispatchToProps = dispatch => ({
  getData: () => dispatch(getData()),
  getToken: (route) => dispatch(getToken(route))
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Home)
