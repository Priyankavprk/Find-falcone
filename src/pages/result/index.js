import React, { Component } from 'react';
import { connect } from 'react-redux';

import Footer from '../../components/Footer';
import Header from '../../components/Header';
import logo from '../../king.png';
import '../App.css';

export class Result extends Component {
  render() {
    if (this.props.data) {
      return (
        <div className="App">
          <Header />
          <header className="App-header">
            <div className="image">
              <img className="icon" src={logo} alt="king"/>
            </div>
            {
              this.props.status === "success" &&
              <div className="titlemain">
                <label> Success ! Congragulations on Finding Falcone. King Shan is mighty pleased.</label><br/>
                <label> Time taken: {Object.values(this.props.data).reduce(function(a, b) { return a + b; }, 0)} </label><br/>
                <label> Planet found: {this.props.successPlanet} </label><br/>
              </div>
            }
            {
              this.props.status === "false" &&
              <div className="titlemain">
                <label> Mission failed... Please try again... </label>
              </div>
            }
            <div className="button">
              <input type="submit" onClick={() => {this.props.history.push('/'); this.props.resetState()}} value="Start Again" />
            </div>
          </header>
          <Footer />
        </div>
      );
    } else {
      return (
        <div className="loading"> Loading ... </div>
      )
    }
  }
}

const mapStateToProps = state => ({
  successPlanet: state.list.successPlanet,
  status: state.list.status,
  data: state.list.data
})

const mapDispatchToProps = dispatch => ({
  resetState: () => dispatch({type: 'RESET_STATE'})
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Result)
