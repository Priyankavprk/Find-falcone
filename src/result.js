import React, { Component } from 'react';

import Footer from './components/footer';
import Header from './components/header';
import logo from './king.png';
import './App.css';

class Result extends Component {
  render() {
    if (this.props.location && this.props.location.state.data) {
      return (
        <div className="App">
          <Header />
          <header className="App-header">
            <div className="image">
              <img className="icon" src={logo} alt="king"/>
            </div>
            {
              this.props.location.state.data.status === "success" &&
              <div className="titlemain">
                <label> Success ! Congragulation on Finding Falcone. King Shan is mighty pleased.</label><br/>
                <label> Time taken: {Object.values(this.props.location.state.time).reduce(function(a, b) { return a + b; }, 0)} </label><br/>
                <label> Planet found: {this.props.location.state.data.planet_name} </label><br/>
              </div>
            }
            {
              this.props.location.state.data.status === "false" &&
              <div className="titlemain">
                <label> Mission failed... Please try again... </label>
              </div>
            }
            <div className="button">
              <input type="submit" onClick={() => {this.props.history.push('/')}} value="Start Again" />
            </div>
          </header>
          <Footer />
        </div>
      );
    } else {
      return (
        <div className="loading"> Loading ... </div>
      )
    }
  }
}

export default Result;
