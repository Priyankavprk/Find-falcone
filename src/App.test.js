import React from 'react';
import ReactDOM from 'react-dom';
import { mount, shallow } from "enzyme";
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { Home } from './pages/home';
import { Result } from './pages/result';
import { Dropdown } from './components/DropDown';
import { Header } from './components/Header';
import { Footer } from './components/Footer';

Enzyme.configure({ adapter: new Adapter() });

it('Home renders without crashing', () => {
  const getDataMock = jest.fn();
  const component = shallow(<Home getData={getDataMock} />);
  expect(component).toMatchSnapshot();
});

it('Result renders without crashing', () => {
  const component = shallow(<Result />);
  expect(component).toMatchSnapshot();
});

it('Header renders without crashing', () => {
  const component = shallow(<Header />);
  expect(component).toMatchSnapshot();
});

it('Footer renders without crashing', () => {
  const component = shallow(<Footer />);
  expect(component).toMatchSnapshot();
});
