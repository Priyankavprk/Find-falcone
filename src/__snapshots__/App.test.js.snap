// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <ForwardRef(Bootstrap(NavbarBrand))
        className="title"
      >
        Mission Find Falcone
      </ForwardRef(Bootstrap(NavbarBrand))>,
      "className": "header",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "Mission Find Falcone",
        "className": "title",
      },
      "ref": null,
      "rendered": "Mission Find Falcone",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "propTypes": undefined,
        "render": [Function],
      },
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "Brand": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "propTypes": undefined,
        "render": [Function],
      },
      "Collapse": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "propTypes": undefined,
        "render": [Function],
      },
      "Text": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "propTypes": undefined,
        "render": [Function],
      },
      "Toggle": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "propTypes": undefined,
        "render": [Function],
      },
      "defaultProps": undefined,
      "propTypes": undefined,
      "render": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ForwardRef(Bootstrap(NavbarBrand))
          className="title"
        >
          Mission Find Falcone
        </ForwardRef(Bootstrap(NavbarBrand))>,
        "className": "header",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Mission Find Falcone",
          "className": "title",
        },
        "ref": null,
        "rendered": "Mission Find Falcone",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "propTypes": undefined,
          "render": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "Brand": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "propTypes": undefined,
          "render": [Function],
        },
        "Collapse": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "propTypes": undefined,
          "render": [Function],
        },
        "Text": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "propTypes": undefined,
          "render": [Function],
        },
        "Toggle": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "propTypes": undefined,
          "render": [Function],
        },
        "defaultProps": undefined,
        "propTypes": undefined,
        "render": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Header renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <ForwardRef(Bootstrap(NavbarBrand))
        className="title"
      >
        Finding Falcone!
      </ForwardRef(Bootstrap(NavbarBrand))>,
      "className": "header",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "Finding Falcone!",
        "className": "title",
      },
      "ref": null,
      "rendered": "Finding Falcone!",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "propTypes": undefined,
        "render": [Function],
      },
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "Brand": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "propTypes": undefined,
        "render": [Function],
      },
      "Collapse": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "propTypes": undefined,
        "render": [Function],
      },
      "Text": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "propTypes": undefined,
        "render": [Function],
      },
      "Toggle": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "propTypes": undefined,
        "render": [Function],
      },
      "defaultProps": undefined,
      "propTypes": undefined,
      "render": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ForwardRef(Bootstrap(NavbarBrand))
          className="title"
        >
          Finding Falcone!
        </ForwardRef(Bootstrap(NavbarBrand))>,
        "className": "header",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Finding Falcone!",
          "className": "title",
        },
        "ref": null,
        "rendered": "Finding Falcone!",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "propTypes": undefined,
          "render": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "Brand": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "propTypes": undefined,
          "render": [Function],
        },
        "Collapse": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "propTypes": undefined,
          "render": [Function],
        },
        "Text": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "propTypes": undefined,
          "render": [Function],
        },
        "Toggle": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "propTypes": undefined,
          "render": [Function],
        },
        "defaultProps": undefined,
        "propTypes": undefined,
        "render": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Home renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home
    getData={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": " Loading ... ",
      "className": "loading",
    },
    "ref": null,
    "rendered": " Loading ... ",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": " Loading ... ",
        "className": "loading",
      },
      "ref": null,
      "rendered": " Loading ... ",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Result renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Result />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": " Loading ... ",
      "className": "loading",
    },
    "ref": null,
    "rendered": " Loading ... ",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": " Loading ... ",
        "className": "loading",
      },
      "ref": null,
      "rendered": " Loading ... ",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
